src/components/views/rooms/wysiwyg_composer/hooks/useInputEventProcessor.ts

```diff
@@ -29,9 +29,14 @@ export function useInputEventProcessor(onSend: () => void): (event: WysiwygEvent
 
             const isKeyboardEvent = event instanceof KeyboardEvent;
             const isEnterPress =
-                !isCtrlEnter && (isKeyboardEvent ? event.key === "Enter" : event.inputType === "insertParagraph");
+                !isCtrlEnter &&
+                (isKeyboardEvent
+                    ? // Ugly but here we need to send the message only if Enter is pressed
+                      // And we need to stop the event propagation on enter to avoid the composer to grow
+                      event.key === "Enter" && !event.shiftKey && !event.ctrlKey && !event.metaKey && !event.altKey
+                    : event.inputType === "insertParagraph");
```

Comment:
Mm, this is making my head hurt somewhat. Use let & some separate if statements to conditionally change it? Or if it must be functional, define `isEnterPress` as the result of an anonymous function?

Reply:
@dbkr I added some extra informations in the PR title and I have split the operation in a dedicated function in multiple booleans.